src/main/java/org/openrewrite/kotlin/Assertions.java

```diff
@@ -99,7 +99,8 @@ public static Consumer<SourceSpec<K.CompilationUnit>> isFullyParsed() {
             new KotlinIsoVisitor<Integer>() {
                 @Override
                 public Space visitSpace(Space space, Space.Location loc, Integer integer) {
-                    if (!space.getWhitespace().trim().isEmpty()) {
+                    String trimmed = space.getWhitespace().trim();
+                    if (!trimmed.isEmpty() && !trimmed.equals("!")) {
```

Comment:
Now that we have this generic check to verify if the printer prints back out whatever was parsed, I am a bit unsure if we still need this `isFullyParsed()` assertion or not. If we still want to keep it, I would change this line to only allow the `!` when we are in a context where this is actually allowed.

Reply:
I think it's fine to remove `isFullyParsed()`. It was easier to find parsing issues before the idempotent print check was added because we add parser error information when it happens. Most of that is lost now, so we can remove the J.Unknown as well.